### PR TITLE
Allow viewing SARIF from query history view

### PR DESCRIPTION
**Pull request by jcreedcmu**
Addresses the SARIF part of https://github.com/github/vscode-codeql/issues/342. Could still use CSV export.

`@github/product-docs-dsp` any need for docs for this?

### CODE REVIEW

File: extensions/ql-vscode/src/query-history.ts [hunk 1]
```diff
@@ -83,6 +83,11 @@ class HistoryTreeDataProvider implements vscode.TreeDataProvider<CompletedQuery>
       arguments: [element],
     };
 
+    // Mark this query history item according to whether it has a
+    // SARIF file so that we can make context menu items conditionally
+    // available.
+    it.contextValue = element.query.hasInterpretedResults() ? 'interpretedResultsItem' : 'rawResultsItem';
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/query-history.ts:89]: Neat.

File: extensions/ql-vscode/src/run-queries.ts [hunk 2]
```diff
@@ -157,15 +157,22 @@ export class QueryInfo {
...
       logger.log("Cannot produce interpreted results since the database does not have a .dbinfo or codeql-database.yml file.");
     }
     return hasMetadataFile;
   }
+
+  /**
+   * Holds if this query actually has produced interpreted results.
+   */
+  hasInterpretedResults(): boolean {
+    return fs.existsSync(this.resultsPaths.interpretedResultsPath);
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/run-queries.ts:174]: Can we make this function async? Otherwise, this is blocking.

**Comment by aeisenberg** [comment]
And looks like the unit tests caught a legitimate issue that would have been hard to find otherwise. Nice. :)

File: extensions/ql-vscode/src/run-queries.ts (hunk 2 above)
Comment (jcreedcmu) [reply extensions/ql-vscode/src/run-queries.ts:174]: Ah, I was afraid the `getTreeItem` had to by synchronous, but it looks like vscode permits it to return a `Promise`. Happy to make it `async` then.

**Comment by jcreedcmu** [comment]
> And looks like the unit tests caught a legitimate issue that would have been hard to find otherwise. Nice. :)

Ah, right, I did leave out the command palette exclusion to see if it would trigger that, then promptly forgot that I had done so. Thankfully, unit tests don't forget :)

**Pull request merged**